### PR TITLE
feat(frontend): IDB functions for Bitcoin address

### PR DESCRIPTION
**Pull request by AntonioVentilii-DFINITY**
# Motivation

We replicate the IDB functions for Bitcoin Mainnet address, as they are used for Ethereum.

### CODE REVIEW

File: src/frontend/src/lib/types/idb.ts [hunk 1]
```diff
@@ -1,4 +1,10 @@
-import type { EthAddress } from '$lib/types/address';
+import type { BtcAddress, EthAddress } from '$lib/types/address';
+
+export interface IdbBtcAddress {
+	address: BtcAddress;
+	createdAtTimestamp: number;
```
Comment (peterpeterparker) [inline src/frontend/src/lib/types/idb.ts:5]: should we have a common interface `IdbAddress` that declares the timestamps and implements it for Btc and Eth?

**Review by peterpeterparker — APPROVED** [review]
LGTm, thx

File: src/frontend/src/lib/types/idb.ts (hunk 1 above)
Comment (AntonioVentilii-DFINITY) [reply src/frontend/src/lib/types/idb.ts:5]: Yes! Agreed, something like:

```typescript
interface IdbAddress<T> {
	address: T;
	createdAtTimestamp: number;
	lastUsedTimestamp: number;
}

export type IdbBtcAddress = IdbAddress<BtcAddress>;

export type IdbEthAddress = IdbAddress<EthAddress>;
```

Comment (peterpeterparker) [reply src/frontend/src/lib/types/idb.ts:5]: Nice

**Pull request merged**